test/AMAMM.t.sol

```diff
@@ -24,81 +25,136 @@ contract AmAmmTest is Test {
+        vm.startPrank(user0);
+        amAmm.bidToken().approve(address(amAmm), 1e18 * K);
+        amAmm.bid(POOL_0, _swapFeeToPayload(0.01e6), 1e18, 1);
+        vm.stopPrank();
 
-    //     vm.prank(user1);
-    //     vm.expectRevert();
-    //     amAmm.withdrawBalance(POOL_0, K * 10e18);
-    // }
+        vm.startPrank(user1);
+        amAmm.bidToken().approve(address(amAmm), 10e18 * K);
+        amAmm.bid(POOL_0, _swapFeeToPayload(0.01e6), 10e18, 1); // New Bid Winner
+        vm.stopPrank();
 
-    // // Test if lower bids can be refunded
-    // function test_bid_refund() external {
-    //     vm.prank(user0);
-    //     amAmm.bid(POOL_0, _swapFeeToPayload(0.01e6), 1e18, K * 1e18, 1);
+        assertEq(amAmm._getDeposit(POOL_0, 1), K * 10e18, "Bid Promoted to Top Bid");
 
-    //     vm.prank(user1);
-    //     amAmm.bid(POOL_0, _swapFeeToPayload(0.01e6), 10e18, K * 10e18, 1); // New Bid Winner
+        skip(10800); //Enter Epoch 3
 
-    //     assertEq(amAmm._getDeposit(POOL_0, 1), K * 10e18, "Bid Promoted to Top Bid");
+        assertEq(amAmm._userBalance(user0), 1e18 * K, "Ensure User bid is still available to withdraw");
 
-    //     skip(10800); //Enter Epoch 3
+        vm.prank(user0); // user0 should have 1e18* k balance and should be able to withdraw
+        amAmm.withdrawBalance(POOL_0, 1e18 * K);
 
-    //     assertEq(amAmm._userBalance(user0), 1e18 * K, "Ensure User bid is still available to withdraw");
+        assertEq(amAmm._userBalance(user0), 0, "Zero Balance after withdrawing");
+    }
 
-    //     vm.prank(user0); // user0 should have 1e18* k balance and should be able to withdraw
-    //     amAmm.withdrawBalance(POOL_0, 1e18 * K);
+    function test_bid_withdraw() external {
+        amAmm.bidToken().mint(address(user0), K * 100e18);
+        amAmm.bidToken().mint(address(user1), K * 100e18);
 
-    //     assertEq(amAmm._userBalance(user0), 0, "Zero Balance after withdrawing");
-    // }
+        vm.startPrank(user0);
+        amAmm.bidToken().approve(address(amAmm), 1e18 * K);
+        amAmm.bid(POOL_0, _swapFeeToPayload(0.01e6), 1e18, 1);
 
-    // function test_bid_withdraw() external {
-    //     vm.prank(user0);
-    //     amAmm.bid(POOL_0, _swapFeeToPayload(0.01e6), 1e18, K * 1e18, 0);
-    //     vm.prank(user0);
-    //     amAmm.bid(POOL_0, _swapFeeToPayload(0.01e6), 2e18, K * 2e18, 0);
+        vm.stopPrank();
 
-    //     vm.prank(user0);
-    //     amAmm.withdrawBalance(POOL_0, K * 1e18);
+        vm.startPrank(user0);
+        amAmm.bidToken().approve(address(amAmm), 2e18 * K);
+        amAmm.bid(POOL_0, _swapFeeToPayload(0.01e6), 2e18, 1);
+        vm.stopPrank();
 
-    //     assertEq(amAmm._userBalance(user0), 0, "Zero Balance after withdrawing");
-    // }
+        vm.prank(user0);
+        amAmm.withdrawBalance(POOL_0, K * 1e18);
+
+        assertEq(amAmm._userBalance(user0), 0, "Zero Balance after withdrawing");
```

Comment:
```suggestion
        assertEq(amAmm._userBalance(user0), originalBalance - 10, "Zero Balance after withdrawing");
```
It's better to test with a variable name as I am not sure about the numbers that you started with 100 and bidding with 1 and then 2; shouldn't it still have 97 remaining?